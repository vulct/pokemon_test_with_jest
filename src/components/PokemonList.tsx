import React, { FC } from "react";
import "./pokemon.css";

interface Props {
  id: number;
  image: string;
  name: string;
}

const PokemonList: FC<Props> = (props) => {
  const { id, image, name } = props;

  return (
    <div className="">
      <section className="pokemon-list-container">
          <p className="pokemon-name">{name}</p>
          <img src={image} alt="pokemon" />
      </section>
    </div>
  );
};

export default PokemonList;
