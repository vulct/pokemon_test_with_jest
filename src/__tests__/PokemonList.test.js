import { shallow } from "enzyme";
import PokemonList from "../components/PokemonList";

describe("PokemonList", function() {
    let wrapper;
    beforeEach(function() {
        wrapper = shallow(<PokemonList {...this.props} />);
    });

    it("should render without crashing", function() {
        expect(wrapper).toMatchSnapshot();
    });
});    