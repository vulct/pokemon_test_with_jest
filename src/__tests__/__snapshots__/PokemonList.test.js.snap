// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonList should render without crashing 1`] = `
<div
  className=""
>
  <section
    className="pokemon-list-container"
  >
    <p
      className="pokemon-name"
    />
    <img
      alt="pokemon"
    />
  </section>
</div>
`;
